dowhy/datasets.py

```diff
@@ -177,7 +184,9 @@ def _compute_y(t, W, X, FD, beta, c2, ce, cfd2):
     data = pd.DataFrame(data, columns=col_names)
     # Specifying the correct dtypes
     if treatment_is_binary:
-        data = data.astype({tname:'bool' for tname in treatments}, copy=False)
+        data = data.astype({tname: "bool" for tname in treatments}, copy=False)
+    elif treatment_is_category:
+        data = data.astype({tname: "category" for tname in treatments}, copy=False)
```

Comment:
This line is important, since pandas detects this type and then does one hot encoding of this column when `get_dummies` is later called